### ISSUE TITLE
Support for Samsung 7 Series NU7670 (UE49NU7670)

### ISSUE THREAD

**Issue by JamiePhonic** [jaruba/ha-samsungtv-tizen#50]
Hello!

I have a Samsung 7 Series NU7670 Tizen based smart TV (Full model #UE49NU7670) but i cant seem to get it working with home assistant in any way other than via the SmartThings integration.

Ports 8001 (HTTP) and 8002 (HTTPS) are open, but i only get either a 401 for the root URL path (192.168.1.100:8001/), or 404 for any for the possible API paths i've seen. (like /api/v2)

I've checked and made sure remote access is allowed in the settings. Attempting to connect with the default "SamsungTV" integrating from the UI just returns "Aborted: Unable to connect to this Samsung TV device."

I'm comfortable with WireShark and the like so happy to do some prodding if needs be.

**Comment by jaruba** [comment]
The NU model should be from 2018, so it should work, the 8001/8002 ports should be for a WebSocket connection, not HTTP(s).

**Comment by JamiePhonic** [comment]
im seeing this in the logs:
```Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 434, in _async_add_entity
    await entity.async_update_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 284, in async_update_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 314, in _async_write_ha_state
    attr = self.capability_attributes
  File "/usr/src/homeassistant/homeassistant/components/media_player/__init__.py", line 781, in capability_attributes
    source_list = self.source_list
  File "/config/custom_components/samsungtv_tizen/media_player.py", line 532, in source_list
    self._gen_installed_app_list()
  File "/config/custom_components/samsungtv_tizen/media_player.py", line 337, in _gen_installed_app_list
    app_list = self._ws.app_list()
  File "/config/custom_components/samsungtv_tizen/websockets.py", line 237, in app_list
    'to': 'host'
  File "/config/custom_components/samsungtv_tizen/websockets.py", line 109, in _ws_send
    self.open()
  File "/config/custom_components/samsungtv_tizen/websockets.py", line 149, in open
    sslopt=sslopt
  File "/usr/local/lib/python3.7/site-packages/websocket/_core.py", line 514, in create_connection
    websock.connect(url, **options)
  File "/usr/local/lib/python3.7/site-packages/websocket/_core.py", line 223, in connect
    options.pop('socket', None))
  File "/usr/local/lib/python3.7/site-packages/websocket/_http.py", line 120, in connect
    sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
  File "/usr/local/lib/python3.7/site-packages/websocket/_http.py", line 189, in _open_socket
    raise error
  File "/usr/local/lib/python3.7/site-packages/websocket/_http.py", line 172, in _open_socket
    sock.connect(address)
TimeoutError: [Errno 110] Operation timed out```

I'm not sure, but could it be a network issue?
My HA server and the TV sit on different subnets. it hasn't been a problem for anything else so far, but i'm wondering if that could be part of the problem? (There arn't any firewall rules blocking communication.)

I'll try moving the TV to the same subnet as the HA server anyway and report back. If that does end up being the case, it might be worth making a note of that somewhere!

**Comment by jaruba** [comment]
No, just set the app_list property in the config, as described in the readme, and that error will go away.

**Comment by JamiePhonic** [comment]
While providing an apps list has stopped that error, all of the controls appear to fail with the same error. In this example, Volume Up, and the entity state displays "Unknown"

``` Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 134, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1234, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1257, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 198, in handle_service
    self._platforms.values(), func, call, required_features
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 402, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 599, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 433, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/media_player/__init__.py", line 719, in async_volume_up
    await self.hass.async_add_job(self.volume_up)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/samsungtv_tizen/media_player.py", line 603, in volume_up
    self.send_command("KEY_VOLUP")
  File "/config/custom_components/samsungtv_tizen/media_player.py", line 446, in send_command
    self._ws.send_key(payload, key_press_delay)
  File "/config/custom_components/samsungtv_tizen/websockets.py", line 181, in send_key
    key_press_delay
  File "/config/custom_components/samsungtv_tizen/websockets.py", line 109, in _ws_send
    self.open()
  File "/config/custom_components/samsungtv_tizen/websockets.py", line 160, in open
    raise exceptions.ConnectionFailure(response)
custom_components.samsungtv_tizen.exceptions.ConnectionFailure: {'event': 'ms.channel.timeOut'}
```

**Comment by jaruba** [comment]
Looks like a timeout, did u accept the authorization pop-ups on the TV? Did u disable HAs default component for Samsung TVs?

**Comment by JamiePhonic** [comment]
I've never received an authorisation popup on the TV from HA. I'm not using the fedault Samsung TV component.
In Settings => General => Device Connection Manager; 
`Access Notification` is set to `First Time Only`
In Settings => General => Network => Expert Settings;
`Power on with Mobile` and `IP Remote` are both `Enabled`

Im running HomeAssistant using a HassOS VM updated to the latest version (OS 3.12, Supervisor 217) if that's of any consequence?

**Comment by JamiePhonic** [comment]
So just to rule it out, I moved my HA server and my TV onto the same subnet to see if that would help.
Turns out, it did! 
Once both were on the same subent, I restarted HA after updating the config with the TV's new IP, Notification appeared on the TV, accepted and everything started working as expected. (volume control, source/app seelction, the lot!)

I'll close this issue off now but it might be worth making a note somewhere that your HA server and TV must be on the same network. (It's not a common setup, i know, but it couldn't hurt to mention)
